File: ios-client/Exametrics/Exametrics/Point.swift
//
//  Point.swift
//  Exametrics
//
//  Created by Vilaine Emilien on 19/05/2016.
//  Copyright © 2016 IMERIR. All rights reserved.
//

import Foundation
import RealmSwift
import Realm

class Point : Object{
    
    // Champs
    private dynamic var _id        : String = ""
    private dynamic var _longitude : String = ""
    private dynamic var _latitude  : String = ""
    private dynamic var _idArea    : String = ""
    
    // Getter & Setter
    func getId() -> String{
        return _id
    }
    
    func setId(id: String) {
        _id = id
    }
    
    func getLongitude() -> Double{
        return Double(_longitude)!
    }
    
    func setLongitude(longitude: Double) {
        _longitude = String(longitude)
    }
    
    func getLatitude() -> Double{
        return Double(_latitude)!
    }
    
    func setLatitude(latitude: Double) {
        _latitude = String(latitude)
    }
    
    func getIdArea() -> String{
        return _idArea
    }
    
    func setIdArea(idArea: String) {
        _idArea = idArea
    }
    
    // Constructeurs
    func setPoint(id: String, longitude: Double, latitude: Double, idArea: String){
        
        _id = id
        _longitude = String(longitude)
        _latitude = String(latitude)
        _idArea = idArea
    }
    
    
}

File: android-admin/exametric-administration/app/src/main/java/com/exametric_administration/classes/classes/Point.java
package com.exametric_administration.classes.classes;

import io.realm.RealmObject;

public class Point extends RealmObject {

    private int idPoint;
    private String longitude;
    private String latitude;
    private int idArea;

    public int GetIdArea() {
        return idArea;
    }

    public void SetIdArea(int idArea) {
        this.idArea = idArea;
    }

    public int GetIdPoint() {
        return idPoint;
    }

    public void SetIdPoint(int idPoint) {
        this.idPoint = idPoint;
    }

    public String GetLongitude() {
        return longitude;
    }

    public void SetLongitude(String longitude) {
        this.longitude = longitude;
    }

    public String GetLatitude() {
        return latitude;
    }

    public void SetLatitude(String latitude) {
        this.latitude = latitude;
    }

    public Point() {}

    public Point(int idPoint, String longitude, String latitude, int idArea) {
        SetIdPoint(idPoint);
        SetLongitude(longitude);
        SetLatitude(latitude);
        SetIdArea(idArea);

    }

}
